File: 169 Majority Element/169majority-element.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        count = Counter(nums)
        res = 0

        for num, cnt in count.items():
            if cnt > len(nums) / 2:
                res = num
        return res

# Time: O(n)
# Space: O(n)

File: 169 Majority Element/169majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        int res = 0, count = 0;

        for (int num: nums) {
            if (count == 0) {
                res = num;
            }
            count += (num == res) ? 1 : -1;
        }
        return res;
    }
};

// Time: O(n)
// Space: O(1)